src/rp2_common/pico_stdio_usb/stdio_usb.c

```diff
@@ -139,8 +175,11 @@ bool stdio_usb_init(void) {
     if (irq_has_shared_handler(USBCTRL_IRQ)) {
         // we can use a shared handler to notice when there may be work to do
         irq_add_shared_handler(USBCTRL_IRQ, usb_irq, PICO_SHARED_IRQ_HANDLER_LOWEST_ORDER_PRIORITY);
+        critical_section_init_with_lock_num(&one_shot_timer_crit_sec, next_striped_spin_lock_num());
     } else {
-        rc = add_alarm_in_us(PICO_STDIO_USB_TASK_INTERVAL_US, timer_task, NULL, true);
+        rc = add_alarm_in_us(PICO_STDIO_USB_TASK_INTERVAL_US, timer_task, NULL, true) >= 0;
+        // we use initialization state of the one_shot_timer_critsec as a flag
+        memset(&one_shot_timer_crit_sec, 0, sizeof(one_shot_timer_crit_sec));
```

Comment:
So AFAICT you're setting `one_shot_timer_crit_sec` to NULL, here, and then elsewhere you're doing `critical_section_is_initialized(&one_shot_timer_crit_sec)`. But given that `critical_section_is_initialized` checks `crit_sec->spin_lock != 0`, won't that lead to a null-pointer dereference? :man_shrugging:

Reply:
setting the contents of one_shot_timer_crit_sec to zero.